src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java

```diff
@@ -3540,6 +3540,32 @@
     public static final int COMMENT_CONTENT =
             GeneratedJavaTokenTypes.COMMENT_CONTENT;
 
+    /**
+     * The {@code record} keyword.  This element appears
+     * as part of a record declaration.
+     **/
+    public static final int LITERAL_RECORD =
+            GeneratedJavaTokenTypes.LITERAL_record;
+
+    /**
+     * A record declaration.
```

Comment:
as we skip ``--OBJBLOCK -> OBJBLOCK` in AST, lets update `A record declaration` to `A record declaration, this implementation is just to avoid parse errors, full support will be at #8267`

Reply:
Fixed and rebased.